sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -385,9 +385,16 @@ object DateTimeUtils {
    * microseconds where microsecond 0 is 1970-01-01 00:00:00Z.
    */
   def instantToMicros(instant: Instant): Long = {
-    val us = Math.multiplyExact(instant.getEpochSecond, MICROS_PER_SECOND)
-    val result = Math.addExact(us, NANOSECONDS.toMicros(instant.getNano))
-    result
+    val secs = instant.getEpochSecond
+    if (secs == Math.floorDiv(Long.MinValue, MICROS_PER_SECOND)) {
+      val us = Math.multiplyExact(secs + 1, MICROS_PER_SECOND)
+      val result = Math.addExact(us, NANOSECONDS.toMicros(instant.getNano) - MICROS_PER_SECOND)
+      result
+    } else {
+      val us = Math.multiplyExact(secs, MICROS_PER_SECOND)
+      val result = Math.addExact(us, NANOSECONDS.toMicros(instant.getNano))
+      result
```

Comment:
Let's remove `result`, and just leave `Math.addExact(us, NANOSECONDS.toMicros(instant.getNano))`. The same is for the above.